src/sentry/services/hybrid_cloud/project_key/impl.py

```diff
@@ -46,3 +47,13 @@ def get_project_key_by_region(
         self, *, region_name: str, project_id: str, role: ProjectKeyRole
     ) -> Optional[RpcProjectKey]:
         return self._get_project_key(project_id=project_id, role=role)
+
+    def get_project_keys_by_region(
+        self, *, region_name: str, project_ids: List[str], role: ProjectKeyRole
+    ) -> List[RpcProjectKey]:
+        project_keys = ProjectKey.objects.filter(
+            project__in=project_ids,
+            roles=F("roles").bitor(role.as_orm_role()),
+            status=ProjectKeyStatus.ACTIVE,
+        )
```

Comment:
This result set could also get very large, and we should have an upper limit on the number of rows we'll serialize.